src/pudl/extract/epacems.py

```diff
@@ -191,26 +191,32 @@ def get_data_frame(self, partition: EpaCemsPartition) -> pd.DataFrame:
 
     def _csv_to_dataframe(
         self,
-        csv_file: Path,
+        csv_path: Path,
         ignore_cols: dict[str, str],
         rename_dict: dict[str, str],
         dtype_dict: dict[str, type],
+        chunksize: int = 100_000,
     ) -> pd.DataFrame:
         """Convert a CEMS csv file into a :class:`pandas.DataFrame`.
 
         Args:
-            csv (file-like object): data to be read
+            csv_path: Path to CSV file containing data to read.
 
         Returns:
-            A DataFrame containing the contents of the CSV file.
+            A DataFrame containing the filtered and dtyped contents of the CSV file.
         """
-        return pd.read_csv(
-            csv_file,
+        chunk_iter = pd.read_csv(
+            csv_path,
             index_col=False,
             usecols=lambda col: col not in ignore_cols,
             dtype=dtype_dict,
-            low_memory=False,
-        ).rename(columns=rename_dict)
+            chunksize=chunksize,
+            low_memory=True,
+            parse_dates=["Date"],
+        )
+        df = pd.concat(chunk_iter)
+        dtypes = {k: v for k, v in dtype_dict.items() if k in df.columns}
+        return df.astype(dtypes).rename(columns=rename_dict)
```

Comment:
This apparently worked, but I'm not sure *why* it worked.  I didn't expect it to for several reasons:
- The CSV is in a zipfile, and we're asking pandas to read directly from the zipfile. Does that mean the entire zipfile has to be decompressed before you can get at any of the data?  Or can you just read the first 100,000 rows of a zipfile (seems unlikely).
- When you concatenate together dataframes with automatically generated categorical columns, the categoricals become objects, because each dataframe is using a different dictionary mapping for the categories internally, so there's no peak memory savings here from using categorical columns. You *could* squeeze that savings out (and I thought we would have to) by explicitly stating the categories in the `CategoricalDtype()` ensuring that all the dataframes have the same categories, *or* by iteratively concatenating the many per-chunk dataframes together one at a time, and explicitly converting the categorical in both the new and already concatenated dataframes dynamically to match, using [union categoricals](https://pandas.pydata.org/docs/reference/api/pandas.api.types.union_categoricals.html)